pyvrp/cpp/search/TwoOpt.cpp

```diff
@@ -75,15 +77,21 @@ Cost TwoOpt::evalBetweenRoutes(Route::Node *U,
     deltaCost += costEvaluator.twPenalty(vTWS.totalTimeWarp());
     deltaCost -= costEvaluator.twPenalty(V->route->timeWarp());
 
-    deltaCost += costEvaluator.loadPenalty(U->cumulatedLoad + V->route->load()
-                                               - V->cumulatedLoad,
-                                           U->route->capacity());
+    // Proposed move appends the segment after V to U, and the segment after U
+    // to V. So we need to make a distinction between the loads at U and V, and
+    // the loads from clients visited after these nodes.
+    auto const uLoad = U->route->loadBetween(0, U->position);
+    auto const uLoadAfter = U->route->load() - uLoad;
```

Comment:
I would rather not use `endDepot` because I want to make that field private. An alternative is
```cpp
    auto const uLoadAfter
        = U->route->loadBetween(U->position + 1, U->route->size() + 1);
```
but that's also more verbose and not necessarily clearer than "total load - everything depot -> client". I'm inclined to leave this as-is.